### PR TITLE
warm up do not work for compiled model

### PR DESCRIPTION
**Pull request by jiqing-feng**
The warm-up for models which applied torch.compile is not working because of different shapes.

The following codes can reproduce
```python
import time
import torch
from transformers import pipeline

model_id = "ProsusAI/finbert"
pipe = pipeline("sentiment-analysis", model=model_id)
pipe.model.forward = torch.compile(pipe.model.forward)

prompt_1 = "This was a masterpiece. Not completely faithful to the books, but enthralling from beginning to end. Might be my favorite of the three."
for i in range(4):
    start = time.time()
    output = pipe(prompt_1)
    end = time.time()
    print(f"time for epoch {i}: {(end-start):.3f} s")

prompt_2 = "I am not happy."
for i in range(4):
    start = time.time()
    output = pipe(prompt_2)
    end = time.time()
    print(f"time for epoch {i}: {(end-start):.3f} s")
```

Same issue also happens on cuda. I think it's an issue of torch.compile.
I suppose we can disable warm-up as most models applied torch.compile in IPEXModel, WDYT? @echarlaix

### CODE REVIEW

**Comment by jiqing-feng** [comment]
Hi @echarlaix @IlyasMoutawwakil . Please review this PR, thanks!

File: optimum/intel/ipex/modeling_base.py [hunk 1]
```diff
@@ -240,11 +240,14 @@ def maybe_apply_torch_compile(self):
         self.compiled = True
 
     def _init_warmup(self):
-        inputs = prepare_jit_inputs(self.model, self.export_feature, False)
-        with torch.no_grad():
-            self.model(**inputs)
-            self.model(**inputs)
-        logger.info("Warm up end")
+        if self.compiled:
+            logger.info("Detected torch.compile is applied, please warm-up by your own case")
```
[the remark this answers is not in the document]
Comment (jiqing-feng) [reply optimum/intel/ipex/modeling_base.py]: Resolved!
